### DISCUSSION TITLE
How to change HTTP Method

### DISCUSSION THREAD

**Discussion by itsme112358** [dotnet/yarp#444] [General] [answered]
Hi!
I was glad to see that there's a .netcore reverse proxy now 👍 
I got it running and it redirects URLs as described.
I can't find out however, how I can configure a Http Method overwrite.
My client is sending a PUT request which needs to be translated to a POST in order for the backend (ERP system with fixed API Methods) to be able to process it.
I tried the following:
```json
        "Match": {
          "Path": "/example/{id}/further/path",
          "Method": "PUT"
        },
        "Transforms": [
          {"PathPattern": "/further('{id}')/DifferentEndpoint"},
          {
            "HttpContext": {
              "Method": "POST"
            }
          }
        ]
```
```json
        "Transforms": [
          {"PathPattern": "/further('{id}')/DifferentEndpoint"},
          {
            "HttpContext": {
              "Method": "POST"
            }
          }
        ]
```
Both configurations start without problems and redirct correctly according to the pattern. I just cannot figure out how to configure the request method so that it gets updated/overwritten/replaced.
In the [documentation](https://microsoft.github.io/reverse-proxy/articles/transforms.html#request-parameters) I found the part about request parameters but no info on how to put it into the appsettings.json.

Any help would be appreciated greatly :)

**Comment by Tratcher** [comment, marked as answer]
While we've set up the basic request transform infrastructure, we haven't yet implemented some of the rules you'd expect like method transforms.

Luckily this scenario is easy to work around in code:
```C#
                endpoints.MapReverseProxy(proxyPipeline =>
                {
                    proxyPipeline.Use((context, next) =>
                    {
                        if (HttpMethods.IsPut(context.Request.Method))
                        {
                            context.Request.Method = HttpMethods.Post;
                        }

                        return next();
                    });
                    proxyPipeline.UseProxyLoadBalancing();
                });
```

Interested in helping add this transform? I've filed https://github.com/microsoft/reverse-proxy/issues/445 for this.

**Reply by itsme112358** [reply]
Hi Chris! I was hoping to solve the issue via configuration instead of code, but your solution is really appreciated!
Also thanks for creating the idea, I'll look into it tomorrow :)
